### PR TITLE
rust: set MacOS env variables only on macOS

### PR DESCRIPTION
**Pull request by iMichka**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/rust.rb [hunk 1]
```diff
@@ -61,21 +61,23 @@ class Rust < Formula
...
+      # Fix build failure for compiler_builtins "error: invalid deployment target
+      # for -stdlib=libc++ (requires OS X 10.7 or later)"
+      ENV["MACOSX_DEPLOYMENT_TARGET"] = MacOS.version
+
+      # Fix build failure for cmake v0.1.24 "error: internal compiler error:
+      # src/librustc/ty/subst.rs:127: impossible case reached" on 10.11, and for
+      # libgit2-sys-0.6.12 "fatal error: 'os/availability.h' file not found
+      # #include <os/availability.h>" on 10.11 and "SecTrust.h:170:67: error:
+      # expected ';' after top level declarator" among other errors on 10.12
+      ENV["SDKROOT"] = MacOS.sdk_path
```
Comment (carlocab) [inline Formula/rust.rb:74]: I think we can remove this? CC @EricFromCanada 

https://github.com/Homebrew/homebrew-core/pull/76780#issuecomment-835362487

Comment (EricFromCanada) [reply Formula/rust.rb:74]: Correct.

**Comment by cho-m** [comment]
It looks like most failures are expected, so may want to check to see if safe to merge
The failures I haven't taken a look at:
- `openimageio`
- `graph-tool` - errors are about `boost`

The failures that are maybe safe to ignore:
- Expected failures `asuka`, `vice`, `gifski`, `ktmpl`, `wasm-pack` based on previous update #73924 and upcoming update https://github.com/Homebrew/homebrew-core/pull/76780#issuecomment-837498763
- Also expected failures `cargo-watch`, `pdfpc`, `freeswitch`
- `gtkspell3` - Fixed in #78388
- `pc6001vx` - C++20 `<version>` issue
- `cargo-audit` - upstream repo merge has broken source tar URL 
- `badtouch` (`authoscope`) - SHA mismatch, probably fixed in #78138

**Comment by carlocab** [comment]
Changes included in #76780.

**Pull request closed without merging**